drivers/hmis/app/graphql/types/forms/form_definition.rb

```diff
@@ -48,6 +51,54 @@ def definition
       )
     end
 
+    def project_matches
+      # Returns one match per project that this form applies to. (See InstanceProjectMatch for match ranking logic.)
+      # This DOES still return a match for a project if the match is overridden by a more specific rule on another form.
+      # For example, if:
+      # - Intake form A has a rule that specifies that it's used for all projects
+      # - Intake form B has a rule that specifies that it's used for all Emergency Shelter projects
+      # - Intake form C has a rule that specifies that it's used for Project X, an ES project
+      # ...then forms A and B would still return a match for Project X.
+
+      instance_scope = object.instances.active
+
+      # Fetching all projects is slow, so do it only once and pass the scope to project_matches below.
+      project_scope = Hmis::Hud::Project.hmis.preload(:organization)
+
+      # Preload funders only if any of these rules are funder-based (micro-optimizing)
+      project_scope = project_scope.preload(:funders) if instance_scope.where.not(funder: nil).exists?
+
+      # Further refine project_scope down to only projects that could be matched by any of the instances
+
+      unless instance_scope.defaults.exists?
+        # If there is a default, then all projects are fair game.
+        # Otherwise, glom together all the projects that might match any of the rules.
+
+        matches_any_project = p_t[:id].in(instance_scope.select { |inst| inst.entity_type == 'Hmis::Hud::Project' }.map(&:entity_id))
+        matches_org = o_t[:id].in(instance_scope.select { |inst| inst.entity_type == 'Hmis::Hud::Organization' }.map(&:entity_id))
```

Comment:
wasn't intentional naming on my part, could be `matches_any_org`, `matches_any_funder` etc